docs/compatibility.html

```diff
@@ -0,0 +1,206 @@
+        <tr>
+            <td>Server</td>
+            <td>4.0.0</td>
+            <td>❌</td>
+            <td>❌</td>
+            <td>✅</td>
+            <td>✅</td>
+        </tr>
+    </table>
+
+    <h4>Server Compatibility</h4>
+    <table>
+        <tr>
+            <th>KRaft Cluster Version</th>
+            <th>Compatibility 4.0 Server (dynamic voter)</th>
+            <th>Compatibility 4.0 Server (static voter)</th>
+        </tr>
+        <tr>
+            <td>before 3.2.x</td>
+            <td>❌</td>
+            <td>❌</td>
+        </tr>
+        <tr>
+            <td>3.3.x</td>
+            <td>❌</td>
+            <td>✅</td>
+        </tr>
+        <tr>
+            <td>3.4.x</td>
+            <td>❌</td>
+            <td>✅</td>
+        </tr>
+        <tr>
+            <td>3.5.x</td>
+            <td>❌</td>
+            <td>✅</td>
+        </tr>
+        <tr>
+            <td>3.6.x</td>
+            <td>❌</td>
+            <td>✅</td>
+        </tr>
+        <tr>
+            <td>3.7.x</td>
+            <td>❌</td>
+            <td>✅</td>
+        </tr>
+        <tr>
+            <td>3.8.x</td>
+            <td>❌</td>
+            <td>✅</td>
+        </tr>
+        <tr>
+            <td>3.9.x</td>
+            <td>✅</td>
+            <td>✅</td>
+        </tr>
+        <tr>
+            <td>4.0.x</td>
+            <td>✅</td>
+            <td>✅</td>
+        </tr>
+    </table>
+    <p>Note: Can’t upgrade server from static voter to dynamic voter, see <a href="https://issues.apache.org/jira/browse/KAFKA-16538">KAFKA-16538</a>.</p>
+
+    <h4>Client Compatibility</h4>
+    <table>
+        <tr>
+            <th>Kafka Version</th>
+            <th>Module</th>
+            <th>Compatibility with Kafka 4.0</th>
+            <th>Key Differences/Limitations</th>
+        </tr>
+        <tr>
+            <td rowspan="3">0.x</td>
+            <td>Client</td>
+            <td>❌ Not Compatible</td>
+            <td>Pre-0.10.x protocols are fully removed in Kafka 4.0 (<a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-896">KIP-896</a>).</td>
+        </tr>
+        <tr>
+            <td>Streams</td>
+            <td>❌ Not Available</td>
+            <td>Kafka Streams introduced in Kafka 0.10.x.</td>
+        </tr>
+        <tr>
+            <td>Connect</td>
+            <td>⚠️ Limited Compatibility</td>
+            <td>Connect introduced in Kafka 0.9.x but lacks support for modern features like IncrementalAlterConfigs and dynamic configs.</td>
+        </tr>
+        <tr>
+            <td rowspan="3">1.x</td>
+            <td>Client</td>
+            <td>⚠️ Limited Compatibility</td>
+            <td>Supports brokers >= 0.10.x. Does not support newer features like idempotent producer or transactional API.</td>
+        </tr>
+        <tr>
+            <td>Streams</td>
+            <td>⚠️ Limited Compatibility</td>
+            <td>Basic Streams features available. Exactly-once semantics unavailable (requires Kafka 2.4.x or later).</td>
```

Comment:
This is confusing. The line compare KS to 4.0 brokers, right? We did add EOS in KS 0.11.0 so what does `(requires Kafka 2.4.x or later)` mean?